src/Css/Style.php

```diff
@@ -3965,159 +4002,185 @@ protected function _compute_size(string $val)
+                    $values = [$this->compute_number($values[0]), 1.0];
+                    break;
+
+                case "scaley":
+                    if ($count > 1) {
+                        return null;
+                    }
+
+                    $name = "scale";
+                    $values = [1.0, $this->compute_number($values[0])];
+                    break;
+
+                // <angle> units
+                case "rotate":
+                    if ($count > 1) {
+                        return null;
+                    }
+
+                    $values = [$this->compute_angle_or_zero($values[0])];
+                    break;
+
+                case "skew":
+                    if ($count > 2) {
+                        return null;
+                    }
+
+                    $values = [
+                        $this->compute_angle_or_zero($values[0]),
+                        isset($values[1]) ? $this->compute_angle_or_zero($values[1]) : 0.0
+                    ];
+                    break;
+
+                case "skewx":
+                    if ($count > 1) {
+                        return null;
+                    }
+
+                    $name = "skew";
+                    $values = [$this->compute_angle_or_zero($values[0]), 0.0];
+                    break;
+
+                case "skewy":
+                    if ($count > 1) {
+                        return null;
+                    }
+
+                    $name = "skew";
+                    $values = [0.0, $this->compute_angle_or_zero($values[0])];
+                    break;
+
+                default:
+                    return null;
+            }
+
+            foreach ($values as $v) {
+                if ($v === null) {
+                    return null;
                 }
             }
+
+            $transforms[] = [$name, $values];
         }
 
         return $transforms;
     }
 
     /**
-     * @param string $computed
-     * @return array
-     *
      * @link https://www.w3.org/TR/css-transforms-1/#transform-origin-property
      */
-    protected function _get_transform_origin($computed)
+    protected function _compute_transform_origin(string $val)
     {
-        //TODO: should be handled in setter
+        $val = strtolower($val);
+        $parts = $this->parse_property_value($val);
+        $count = \count($parts);
 
-        $values = preg_split("/\s+/", $computed);
+        if ($count === 0 || $count > 3) {
+            return null;
+        }
 
-        $values = array_map(function ($value) {
-            if (\in_array($value, ["top", "left"], true)) {
-                return 0;
-            } elseif (\in_array($value, ["bottom", "right"], true)) {
-                return "100%";
-            } else {
-                return $value;
-            }
-        }, $values);
+        $v1 = $parts[0];
+        $v2 = $parts[1] ?? "center";
+        [$x, $y] = $this->computeBackgroundTransformPosition($v1, $v2);
```

Comment:
I don't feel too strong about this, but should we extract the logic of this method? Something like `_compute_transform_position`? `computeBackgroundTransformPosition` would then just be a passthrough to that method, but then the naming would be better aligned.

Reply:
I am not sure I follow? Trying to shorten the name of the method made it more confusing than necessary probably; renamed it to `computeBackgroundPositionTransformOrigin` to make it clearer that it is simply common logic for both properties.

Reply:
Looks like I misread things a bit. The naming confused me since it's a mash up of both property names. I was thinking it was meant for background position computation and was reused. I guess I don't really have a good naming suggestion then.